### PR TITLE
Spring Cloud Stream Reactor support.

### PR DESCRIPTION
**Pull request by joshlong**
Adds support for the reactor spring cloud stream binder.

### CODE REVIEW

File: initializr-service/src/main/resources/application.yml [hunk 1]
```diff
@@ -971,6 +971,17 @@ initializr:
           versionRange: 1.3.0.RELEASE
           groupId: org.springframework.cloud
           artifactId: spring-cloud-stream-binder-kafka
+        - name: Stream Reactor
+          id: cloud-stream-reactive
+          description: Messaging microservices with Reactor. Typically used with a Spring Cloud Stream binder implementation.
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: >  Typically used with a Spring Cloud Stream binder implementation.

What does that mean. Does this work standalone or does this require another dependency?

File: initializr-service/src/main/resources/application.yml [hunk 2]
```diff
@@ -971,6 +971,17 @@ initializr:
...
+          versionRange: 1.3.0.RELEASE
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: That looks fishy.

File: initializr-service/src/main/resources/application.yml [hunk 3]
```diff
@@ -971,6 +971,17 @@ initializr:
...
+          groupId: org.springframework.cloud
+          artifactId: spring-cloud-stream-reactive
+          links:
+            - rel: reference
+              href: https://docs.spring.io/spring-cloud-stream/docs/current/reference/htmlsingle/#_reactive_programming_support
+              description: Reactive Programming support for Spring Cloud Stream
+
```
Comment (snicoll) [inline initializr-service/src/main/resources/application.yml]: I'd like a reference to a sample that demonstrates what you can do with this.

File: initializr-service/src/main/resources/application.yml (hunk 1 above)
Comment (joshlong) [reply initializr-service/src/main/resources/application.yml]: it requires another dependency. You must choose a Spring Cloud Stream binder implementation like Kafka or RabbitMQ

File: initializr-service/src/main/resources/application.yml (hunk 2 above)
Comment (joshlong) [reply initializr-service/src/main/resources/application.yml]: I don't understand. What's "fishy"?

**Comment by snicoll** [comment]
Duplicate of #557

**Comment by snicoll** [comment]
@joshlong we already discussed this entry in another issue you've created.

File: initializr-service/src/main/resources/application.yml (hunk 3 above)
Comment (joshlong) [reply initializr-service/src/main/resources/application.yml]: [Here ya go](https://github.com/joshlong/bootiful-reactive-microservices/blob/master/finchley/reservation-service/src/main/kotlin/com/example/reservationservice/ReservationServiceApplication.kt#L28). In a normal Spring Cloud Stream application I can say 

```
@StreamListener("incomingMessageChannel") 
public void incoming( Message<String> msg) { 
 // handle a new message 
}
```

or:

```
@StreamListener("incomingMessageChannel") 
public void incoming( String msg) { 
 // handle a new message 
}
```

and messages will be delivered to the handler method from whatever message broker you've specified in choosing your Spring Cloud Stream binder. With the new support represented by this checkbox, we can have the incoming values delivered as a Reactor `Flux<T>`. Your method is called once and all values are delivered as new values in the `Flux<T>` stream. This makes it trivial to do things like windowing over values, and integrating with other reactive `Publisher<T>` types (such as those that `WebClient` produces).

**Pull request closed without merging**